internal/querycoordv2/meta/resource_manager.go

```diff
@@ -482,6 +486,55 @@ func (rm *ResourceManager) HandleNodeStopping(node int64) {
 	)
 }
 
+func (rm *ResourceManager) HandleNodeSuspend(node int64) {
+	rm.rwmutex.Lock()
+	defer rm.rwmutex.Unlock()
+
+	if _, ok := rm.suspendNode2RG[node]; ok {
+		return
+	}
+
+	rg := rm.getResourceGroupByNodeID(node)
+	if rg == nil {
+		return
+	}
+
+	rm.suspendNode2RG[node] = rg.GetName()
+	rm.incomingNode.Remove(node)
+	rgName, err := rm.unassignNode(node)
+	log.Info("HandleNodeSuspend: remove node from resource group",
+		zap.String("rgName", rgName),
+		zap.Int64("node", node),
+		zap.Error(err),
+	)
+}
+
+func (rm *ResourceManager) HandleNodeResume(node int64) error {
+	rm.rwmutex.Lock()
+	defer rm.rwmutex.Unlock()
+
+	rgName, ok := rm.suspendNode2RG[node]
+	if !ok {
+		return nil
+	}
+	delete(rm.suspendNode2RG, node)
+
+	if rm.groups[rgName] == nil {
+		log.Info("HandleNodeResume: node's origin rg has been removed, add it to incoming node", zap.String("rgName", rgName))
+		rm.incomingNode.Insert(node)
+		return nil
+	}
+
+	if err := rm.transferNode(rgName, node); err != nil {
```

Comment:
In my opinion, call `transfernode` out of recover logic is not a good choice.
HandleNodeResume is same operation with HandleNodeUp,
HandleNodeSuspend is same operation with HandleNodeDown.
Node state is keep by node manager, resource group manager didn't keep any state of node, just put in or pop out.